integration/spark/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala

```diff
@@ -766,13 +766,13 @@ class CarbonSpark2SqlParser extends CarbonDDLSqlParser {
       throw new MalformedCarbonCommandException("Invalid table properties")
     }
     if (options.isBucketingEnabled) {
-      if (options.bucketNumber.toString.contains("-") ||
-          options.bucketNumber.toString.contains("+") ||  options.bucketNumber == 0) {
+      if (options.bucketNumber.isEmpty ||
+          options.bucketNumber.get.toString.contains("-") || options.bucketNumber.get == 0) {
```

Comment:
Just keep `options.bucketNumber.get <= 0`

it can avoid two condition check and redundant string conversion

Reply:
Done